### PR TITLE
LF-3464: Repeat crop plan | Update UI/UI per discussion

### PR DESCRIPTION
**Pull request by SayakaOno**
**Description**

- subtext describing the original date and its relationship to the management plan.
   eg. _“<due_date> is the earliest task due date from the original crop plan.”_ 
- update `getOccurrencesRuleOptions` function not to duplicate existing plan (if Beginning date is original date and weekday is unchanged from original weekday, the beginning date will not be included in start dates)
- update `calculateMonthlyOptions` to show one more option (4th weekday) when the selected date is both the 4th week of the month and the last

Other issues that are addressed...
- handle timezone with RRule. (It would be great if you could help me with the JSDoc, it was hard to describe...)
- update `getAndSetMonthlyOptions` function to show the selected option when coming back from the confirmation view (@kathyavini could you check if I didn't break anything?)
- fix the scrollbar visibility issue (overflow-y is updated to `auto`)
  This was my setting! ⬇️ 
Screenshot 2023-07-27 at 11 29 02 PM

~~Remaining task:~~ 
~~Crop name validation should be moved to onBlur if possible - can I just change mode: onChange to `onBlur`?~~ 

NOTE: I thought I introduced a new bug on the date input validation that "required" error is shown even though date is there, but it was an existing bug.
Here is the ticket: https://lite-farm.atlassian.net/browse/LF-3480

Jira link: https://lite-farm.atlassian.net/browse/LF-3464

**Type of change**

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**How Has This Been Tested?**

- [ ] Passes test case
- [x] UI components visually reviewed on desktop view
- [ ] UI components visually reviewed on mobile view
- [ ] Other (please explain)

**Checklist:**

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] The precommit and linting ran successfully
- [ ] I have added or updated language tags for text that's part of the UI
- [ ] I have added "MISSING" for all new language tags to languages I don't speak
- [ ] I have added [the GNU General Public License](https://lite-farm.atlassian.net/l/cp/BT0Dd7WW) to all new files

### CODE REVIEW

**Review by kathyavini — COMMENTED** [review]
Great job at the end of a looooong sprint week!!! 🎉 

The date calculation stuff looks 👌 -- awesome!!

Just left a comment about the React Select dropdown selection change. 

Edit: And cool find about the scrollbar! I had no idea that was a setting!

File: packages/webapp/src/components/RepeatCropPlan/index.jsx [hunk 1]
```diff
@@ -119,16 +126,16 @@ export default function PureRepeatCropPlan({
     }
 
     const getAndSetMonthlyOptions = async () => {
+      // Utility function uses rrule to generate natural language strings
+      const options = await calculateMonthlyOptions(planStartDate, repeatFrequency);
+
       // Store which pattern is currently selected
-      const currentSelection = monthlyOptions?.findIndex(
+      const currentSelection = options?.findIndex(
```
[the remark this answers is not in the document]
Comment (kathyavini) [reply packages/webapp/src/components/RepeatCropPlan/index.jsx:133]: Ah sorry I just took a closer read at your PR text! Okay I see why you changed it!

If both can't be saved (the user selection as the starting date is changed vs the user selection when returning from the next page), I'm okay with going this way and only saving the selection after return from confirmation.

[the remark this answers is not in the document]
Comment (SayakaOno) [reply packages/webapp/src/components/RepeatCropPlan/index.jsx]: Thank you Joyce!
I'm sorry I missed this nice comment _"Store which pattern is currently selected"_ and misread your code! I thought it was to keep the exact same option not "pattern".

And thank you so much for sharing the last piece of code! (I combined if-else!)

File: packages/webapp/src/components/RepeatCropPlan/index.jsx [hunk 2]
```diff
@@ -119,16 +126,17 @@ export default function PureRepeatCropPlan({
...
 
     const getAndSetMonthlyOptions = async () => {
-      // Store which pattern is currently selected
-      const currentSelection = monthlyOptions?.findIndex(
+      // Utility function uses rrule to generate natural language strings
+      const options = await calculateMonthlyOptions(planStartDate, repeatFrequency);
+
+      // If already on this screen, store which pattern is currently selected.
+      // When returning from next screen, store the selected option
+      const currentSelection = [monthlyOptions.length ? monthlyOptions : options]?.findIndex(
```
Comment (kathyavini) [inline packages/webapp/src/components/RepeatCropPlan/index.jsx:134]: Nice! This is way more succinct than what I wrote!

I think it has to be parentheses and not brackets though or it doesn't work:

`(monthlyOptions.length ? monthlyOptions : options)?.findIndex(...`

**Review by kathyavini — APPROVED** [review]
Let's fix the `findIndex` later 😄 Tiny change only anyway.

**Review by iperdomo — APPROVED** [review]

File: packages/webapp/src/components/RepeatCropPlan/index.jsx (hunk 2 above)
Comment (SayakaOno) [reply packages/webapp/src/components/RepeatCropPlan/index.jsx:134]: Oh no! What was I doing 🤦‍♀️🤦‍♀️🤦‍♀️ Thank you so much for catching it!!!

**Pull request merged**